src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx

```diff
@@ -143,10 +148,10 @@ export const TemplatesCatalogDrawerCreateForm: FC<TemplatesCatalogDrawerCreateFo
           queryParams: {
             dryRun: 'All',
           },
-        }).then(async (processedTemplate) => {
+        }).then(async (processedTemplate: V1Template) => {
           const vmObject = getTemplateVirtualMachineObject(processedTemplate);
 
-          const updatedVM = produce(vmObject, (vmDraft) => {
+          const updatedVM = produce(vmObject, (vmDraft: V1VirtualMachine) => {
```

Comment:
same as above

Reply:
?

Reply:
It is not recommended to add types that TS can easily figure out. Type annotation vs. type inference. Type inference is over annotation, in TS.

Reply:
it is true about local vars, function signatures should be explicit. 
basically, it makes the code more readable for functions and return values. instead of the developer trying to figure out the type using IDE.

https://www.fun4code.com/typescript-explicit-vs-implicit/
https://medium.com/@ThiliniWathsala/explicit-vs-implicit-type-in-typescript-5727b5cbde

Reply:
Whole course on Udemy about TS I underwent recently was telling me how and why we should prioritize type inference over type annotation. So it is opinion against opinion. Everything will always have some pros and cons. We can argue about what is better forever while our customers are experiencing serious bug I am trying to fix in this PR. I can see this is getting too far and I regret I did more in this PR than it was necessary to fix the bug.

Reply:
you're right that arguing over it is taking too long.

Using "any" as an explicit type or not giving functions signatures the right type if not right. I can recommend watching this part of the course you took a second time, as I'm pretty sure the speaker was referring to local variables or very simple use of functions. Another thing is that 99% of cases there is a difference between theory in course and practice in real-life coding, maybe this is one of those cases referring to the course you watch. 

Any way please remove your WIP so we can get it merged. Thanks

Reply:
> Using "any" as an explicit type or not giving functions signatures the right type is not right. I can recommend watching this part of the course you took a second time, as I'm pretty sure the speaker was referring to local variables or very simple use of functions. Another thing is that 99% of cases there is a difference between theory in course and practice in real-life coding, maybe this is one of those cases referring to the course you watch.

Well, did, and no, I still insist on what I've written earlier, at least for now. And that course was recommended and paid by my employer, they should know what they are doing.

> Any way please remove your WIP so we can get it merged. Thanks

Yes, I'll do, also will remove all the types improvements I've included in this PR. I'll better implement them separately, via followup PR, once we make sure about what and where. It's not good to mix bug fixes with other improvements or refactorings anyway. Thank you for your feedback, opinions and also links to the articles. Of course nothing is black or white, it depends. And sometimes it happens that we deal with something too much, with such details that at the end of the day really don't matter so much.